packages/iov-ethereum/src/parse.ts

```diff
@@ -0,0 +1,21 @@
+import { TokenTicker } from "@iov/bcp-types";
+
+import { constants } from "./constants";
+
+export interface AmountFields {
+  readonly whole: number;
+  readonly fractional: number;
+  readonly tokenTicker: TokenTicker;
+}
+
+export class Parse {
+  public static ethereumAmount(total: string): AmountFields {
+    const totalLenght = total.length;
+    const sigFigs = constants.primaryTokenSigFigs;
+    return {
+      whole: Number(total.substring(0, totalLenght - sigFigs).replace(/^0+/, "")) || 0,
+      fractional: Number(total.substring(totalLenght - sigFigs, totalLenght).replace(/^0+/, "")) || 0,
```

Comment:
When you use slice() instead of substring, whis becomes much more readable. slice() supports negative or unset ends, e.g.

`total.substring(0, -sigFigs)`, `total.substring(-sigFigs)`

Reply:
Number() does not check for integer. Could you wrap string->number into `Int53.fromString(x).toNumber()`? ...which does not work for numbers > 53 bit

Reply:
Then better `Number.parseInt(x, 10)`